students/elmar_m/lesson04/mailroom.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+'''
+file: mailroom.py
+elmar_m / 22e88@mailbox.org
+Lesson04: Mailroom Exercise Part 2
+'''
+
+import time
+import collections
+
+
+main_p = '\n> Main menu. Options: t == thankyou | r == report | q == quit program\n'
+sub_p = '\n>> Sub menu "thankyou". Options: l == list current donors | a == add donation / donor | m == send mail to every donor | x == exit submenu\n'
+
+donors = collections.defaultdict(list)
+ 
+donors['bill'] = [2000, 7.5, 950000]
+donors['steve'] = [5.5, 234000, 928]
+donors['donald'] = [657, 234, 28.57, 90456]
+donors['angie'] = [2, 99, 297765, 47, 28346]
+donors['kim'] = [38982, 66.23, 9856, 0.1]
+
+
+def writefile(name, content):
+    '''
+    Create mail files
+    
+        ARGS:
+    name: filename / name of donor
+    content: mail body text
+    '''
+    ts = time.strftime('%Y%m%d-%H%M%S')
+    fname = '{}.{}.txt'.format(name, ts)
+    f = open(fname, 'w')
+    f.write(content)
+    f.close
+
+
+def mail():
+    '''
+    Create mail body text for every donor
+    '''
+    for i in donors:
+        # print('Dear {}, thank you very much for your donation of {} dollars.\n'.format(i, donors[i][-1]))
+        text = '''Dear {},\n 
+    Thank you very much for your very kind donation of ${}.\n
+    It will be put to very good use.\n
+        Sincerely,
+             -The Team'''.format(i, donors[i][-1])
+        writefile(i, text)
+        print('mailfile created for', i, 'in your current working directory')
+
+
+def thankyou():
+    '''
+    Show list of known donors, add new donor to list, 
+    add new donation to donor and print letter of thanks.
+    '''
+    menu(sub_p, sub_d)
+
+
+def list_donors():
+    '''
+    Print list of donors
+    '''
+    print('\n'.join(donors))
+    
+
+def add():
+    '''
+    Add new donation and / or donor
+    '''
+    dname = input('>> Please give donor name: ')
+    if dname in donors:
+        print('>>', dname, 'already in list')
+        add_amount(dname)
+    elif not dname in donors:
+        print('>>', dname, 'not in list, adding it ')
+        add_amount(dname)
+
```

Comment:
this should really be if/else and you can take the `add_amount` outside of those blocks:
```
if dname in donors:
    print("already in list")
else:
    print("not in list")
add_amount(dname)
```

even shorter (if you wanted):
```
print(">> {dname} {msg}".format(dname=dname, msg="already in list" if dname in donors else "not in list, adding"))
add_amount(dname)

```